libraries/AP_IBus_Telem/AP_IBus_Telem.h

```diff
@@ -0,0 +1,162 @@
+#define IBUS_SENSOR_TYPE_RPM                  0x07 // Throttle value (in 0.01, so 1200 is 12.00%)
+#define IBUS_SENSOR_TYPE_COMPASS_HEADING      0x08 // Heading (0-360 degrees)
+#define IBUS_SENSOR_TYPE_CLIMB_RATE           0x09 // Climb rate (cm/s)
+#define IBUS_SENSOR_TYPE_COG                  0x0a // Course over ground (centidegrees, so 27015 is 270.15 degrees)
+#define IBUS_SENSOR_TYPE_GPS_STATUS           0x0b // GPS status (2 values: fix type, and number of satellites)
+#define IBUS_SENSOR_TYPE_ACC_X                0x0c // Acc X (cm/s)
+#define IBUS_SENSOR_TYPE_ACC_Y                0x0d // Acc Y (cm/s)
+#define IBUS_SENSOR_TYPE_ACC_Z                0x0e // Acc Z (cm/s)
+#define IBUS_SENSOR_TYPE_ROLL                 0x0f // Roll (centidegrees)
+#define IBUS_SENSOR_TYPE_PITCH                0x10 // Pitch (centidegrees)
+#define IBUS_SENSOR_TYPE_YAW                  0x11 // Yaw (centidegrees)
+#define IBUS_SENSOR_TYPE_VERTICAL_SPEED       0x12 // Vertical speed (cm/s)
+#define IBUS_SENSOR_TYPE_GROUND_SPEED         0x13 // Speed (cm/s)
+#define IBUS_SENSOR_TYPE_GPS_DIST             0x14 // Distance from home (m)
+#define IBUS_SENSOR_TYPE_ARMED                0x15 // Armed / unarmed (1 = armed, 0 = unarmed)
+#define IBUS_SENSOR_TYPE_FLIGHT_MODE          0x16 // Flight mode
+#define IBUS_SENSOR_TYPE_ODO1                 0x7c // Odometer1
+#define IBUS_SENSOR_TYPE_ODO2                 0x7d // Odometer2
+#define IBUS_SENSOR_TYPE_SPEED                0x7e // Speed km/h
+#define IBUS_SENSOR_TYPE_ALT_FLYSKY           0xf9 // FlySky-specific altitude (metres)
+
+// 4-byte values
+#define IBUS_SENSOR_TYPE_TEMPERATURE_PRESSURE 0x41 // Combined temperature & pressure value
+#define IBUS_SENSOR_TYPE_GPS_LAT              0x80 // WGS84 in degrees * 1E7
+#define IBUS_SENSOR_TYPE_GPS_LNG              0x81 // WGS84 in degrees * 1E7
+#define IBUS_SENSOR_TYPE_GPS_ALT              0x82 // GPS (cm)
+#define IBUS_SENSOR_TYPE_ALT                  0x83 // Alt (cm)
+#define IBUS_SENSOR_TYPE_ALT_MAX              0x84 // MaxAlt (cm)
+
+// i-BUS vehicle modes
+#define IBUS_VEHICLE_MODE_STAB    0
+#define IBUS_VEHICLE_MODE_ACRO    1
+#define IBUS_VEHICLE_MODE_AHOLD   2
+#define IBUS_VEHICLE_MODE_AUTO    3
+#define IBUS_VEHICLE_MODE_GUIDED  4
+#define IBUS_VEHICLE_MODE_LOITER  5
+#define IBUS_VEHICLE_MODE_RTL     6
+#define IBUS_VEHICLE_MODE_CIRCLE  7
+#define IBUS_VEHICLE_MODE_PHOLD   8
+#define IBUS_VEHICLE_MODE_LAND    9
+#define IBUS_VEHICLE_MODE_UNKNOWN 255 // Must be positive and 0 is already used; out of range blanks the value
+
+class AP_IBus_Telem
+{
+public:
+    AP_IBus_Telem() {}
+
+    CLASS_NO_COPY(AP_IBus_Telem);
+
+    void init(); 
+
+    typedef struct {
+        uint8_t sensor_type;   // Sensor type (IBUS_SENSOR_TYPE_* above)
+        uint8_t sensor_length; // Data length for defined sensor (can be 2 or 4 bytes)
+    } SensorDefinition;
+
+    typedef struct {
+        uint8_t ap_mode;
+        uint8_t ibus_mode;
+    } ModeMap;
```

Comment:
make these private

Reply:
This could be me lacking C++ knowledge here, but if I make them private I'm then told they're inaccessible when using them as types in the .cpp file (as they're being used outside class context, I expect). I could move them out of the class entirely? Or is there a better way?

Reply:
Possibly that's what you mean here? https://github.com/ArduPilot/ardupilot/pull/27695#issuecomment-2258260241

Reply:
Never mind - it would require additional patches in the cpp file to change

```
   98 | static const AP_IBus_Telem::SensorDefinition sensors[] {
```
to
```
   98 | static const AP_IBus_Telem::SensorDefinition AP_IBus_Telem::sensors[] {
```

and the addition in the header of `sensors`.

The definitions could also be moved to the `cpp` file, of course, but we're starting to get lost in the weeds a bit.

We'll just forget I mentioned this possibility :-)

Reply:
This'll be burning in my mind now so I'm sure I'll come back to it for a future iteration 🙂